search/django/weedid/utils.py

```diff
@@ -137,6 +137,17 @@ def remove_entity_local_record(user_id, upload_id):
             os.remove(dir_path)
 
 
+def atomic_copy(repository_dir, download_dir, tmp_dir, upload_id):
```

Comment:
Technically, to improve atomicity, any partially completed copying should be rolled back if an exception occurs (but this won't handle a case of machine failure).